backtrader/feeds/psqlfeed

```diff
@@ -0,0 +1,86 @@
+# !/usr/bin/env python
+# -*- coding: utf-8; py-indent-offset:4 -*-
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+
+from backtrader import DataBase, date2num
+import datetime as dt
+import urlparse
+import psycopg2
+
+class PsqlDatabase(DataBase):
+    '''
+    Fetches data from PSQL, and wraps it into a Feed consumable by cerebro
+    takes url connection string in form of :
+        postgres://{user}:{pw}@{host}:{port [optional]}/{database}
+
+    must also pass a datestart and a table name conforming to a similar schema as following:
+            ticker        char(5),
+            date          date,
+            high          numeric(10,4),
+            low           numeric(10,4),
+            open          numeric(10,4),
+            close         numeric(10,4),
+            volume        integer,
+            unique (ticker, date)
+    '''
+
+    params = (
+        ('url', None),
+        ('dataname', None),
+        ('table', None),
+        ('startdate', None),
+        ('ticker', None),
+    )
+
+    def start(self):
+        try:
+            cursor = self.connect_db()
+        except Exception as e:
+            print('Failed to establish connection to PSQL: {}'.format(e))
+            return
+
+        query = ("""SELECT date, high, low, open, close, volume """
+                 """FROM {table} """
+                 """WHERE ticker = '{ticker}' """
+                 """AND date >= '{startdate}' """
+                 """ORDER BY date asc""".format(
+            table=self.p.table,
+            ticker=self.p.ticker,
+            startdate=dt.datetime.strftime(self.p.startdate, '%Y-%m-%d')
+        ))
+
+        try:
+            cursor.execute(query)
+        except Exception as e:
+            print('PSQL query failed: {}'.format(e))
+
+        self.price_rows = cursor.fetchall()
```

Comment:
self.price_rows will stay in memory till the end of the life of Cerebro engine. During optimization Cerebro will try to preload (if ordered) the data in the main process -  all the data will be copied from the self.price_rows to the internal data feed buffers -  so now two copies of the day will stay in memory.  Once the optimization will start, multiple processes will be forked from the main one - effectively copying all this data unnecessarily for each process, increasing the memory pressure even more.

Suggest to support the 'preload' method and delete the self.price_row at the end of preload.

Reply:
that seems useful!

`    def preload(self):
        super(PsqlDatabase, self).preload()


        # preloaded - no need to keep the price_rows as data is stored in the cerebro's Lines
        self.price_rows = None
       `

how does that look to you? i don't have a very strong understanding of the preload method

Reply:
LGTM

Reply:
added!